src/test/java/com/uid2/shared/auth/RotatingClientKeyProviderTest.java

```diff
@@ -0,0 +1,32 @@
+package com.uid2.shared.auth;
+
+import com.uid2.shared.cloud.EmbeddedResourceStorage;
+import com.uid2.shared.store.CloudPath;
+import com.uid2.shared.store.reader.RotatingClientKeyProvider;
+import com.uid2.shared.store.scope.GlobalScope;
+import io.vertx.core.json.JsonObject;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.*;
+
+public class RotatingClientKeyProviderTest {
+    @Test
+    public void testGetOldestClientKeyBySiteId() throws Exception {
```

Comment:
A test for when site does not have any keys?

Reply:
Test case added as suggested.